Create logs.md (#89)
AdguardTeam/KnowledgeBaseVPN: docs/adguard-vpn-browser-extension/solving-problems/logs.md

---
title: How to collect and send logs
sidebar_position: 1
---

If you encounter a problem when using AdGuard VPN Browser Extension, you can inform us about it by sending application logs.

## Collecting and sending standard logs

> By default, AdGuard VPN Browser Extension uses the Default logging level, that is, the basic collection of data about running processes of the app. To send these logs, follow the next steps:

Reviewer: Actually, in the VPN extension, there is only one logging level

1. Open AdGuard VPN Browser Extension, if possible, repeat the actions that led to the error. Note the exact time when this error occurred.

2. Open *Settings* by clicking the hamburger menu icon (☰) → *Support* → *Report a bug*.

> 3. In the opened form, enter your email address for feedback and describe the error found, including the time when this error occurred. If you can't reproduce the problem, specify as accurately as possible when it last occurred.

Reviewer 2: No need to enter an email. It's filled automatically (but it's possible to change email to another one if needed)

4. There is a check mark next to *Include the diagnostic report in the message*, which means that when you send a report, you also send logs.

> ## Collecting and sending extended logs

Reviewer: in the nearest version of the extension, we would add an export logs button

with the help of this button it would be possible to export logs if there are no possibilities to open the settings page, and without the necessity to open the background page of the browser.

<img width="471" alt="image" src="https://user-images.githubusercontent.com/17272769/218463285-82a6e99a-535c-4fb7-b2eb-1bffdb21f8e1.png">

In most cases, the Default logging level is sufficient to trace down possible bugs. But there are cases when more detailed technical information about the device and connections is required, and then our support team will ask you to collect Debug logs. To do this, follow these steps:

**Chrome**

1. Go to `chrome://extensions`
2. Turn on *developer mode*

![The developer mode *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/dev_mode.png)

3. Click `background.html`

![Background *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/backgroung.png)

4. Open the *Console tab*

![The Console tab *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/console.png)

5. Open the context menu and click *Save as…*

![Save as *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/save.png)

> Send the file with logs to `devteam@adguard.com` with the time of playback. Another option is to add the file to your Google Drive and give an access to `devteam@adguard.com`. 

Reviewer 2: User might not understand the "Google Drive" part. Simply giving us the access doesn't mean we'll be able to see the logs or understand what issue it refers to. I suggest giving an explanation that it's mostly used when the problem is already issued (on github or support) and user should share the link in that ticket.

---
title: How to collect and send logs
sidebar_position: 1
---

> There are several choices for collecting AdGuard VPN Browser extension logs, but we want to draw your attention to the fact that whichever way you go, the same data will be collected. In the extension, logs are not divided into standard and debug ones. And different options for collecting logs are provided for those cases when a problem you encounter prevents you from using one or another instruction.

Author: To avoid repeating "ways-way"

1.  Open AdGuard VPN Browser extension, if possible, repeat the actions that led to the error. Note the exact time when this error occurred.
2.  Open *Settings* by clicking the hamburger menu icon (☰) → *Support* → *Report a bug*.
3.  In the opened form, leave an automatically inserted email address or enter another one and describe the error found, including the time when this error occurred. If you can't reproduce the problem, specify as accurately as possible when it last occurred.
4.  Make sure that there is a check mark next to *Include the diagnostic report in the message*, and tap *Submit*. This way, you will send logs along with the bug report.  

## Collecting and sending logs via the *Export logs* button

1.  Open AdGuard VPN Browser extension, if possible, repeat the actions that led to the error. Note the exact time when this error occurred. 
2.  Right-click the AdGuard VPN Browser extension icon. 
3.  In the opened menu click *Export Logs*.
4.  As a result, the logs file will be saved to your device in `txt` format.

Now that you have collected the logs, you need to hand them over to our development team. To do this: 

> 1.  Report a bug on GitHub. Detailed instructions on which case to create a problem on GitHub and how to do it can be found in [this article](). 

Reviewer 3: ```suggestion
1.  Report a bug on GitHub. Detailed instructions on creating an issue on GitHub can be found in [this article](). 
```

2.  Send an archive with logs and playback time to `devteam@adguard.com` and attach a link to your issue on GitHub or add an archive with logs and playback time to Google Drive and share it with `devteam@adguard.com`. Attach the Google Drive link to your issue on GitHub.

## Collecting and sending logs via browser developer's tools

### Chrome

> 1.  Open AdGuard VPN Browser Extension, if possible, repeat the actions that led to the error. Note the exact time when this error occurred. 

Reviewer 3: AdGuard VPN Browser Extension here, AdGuard VPN Browser extension above

2.  Go to `chrome://extensions`
3.  Turn on *developer mode*

    ![The developer mode *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/dev_mode.png)

4.  Click `background.html`

    ![Background *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/backgroung.png)

5.  Open the *Console tab*

    ![The Console tab *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/console.png)

6.  Open the context menu and click *Save as…*

    ![Save as *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/save.png)

Now that you have collected the logs, you need to hand them over to our development team. To do this: 

> 1.  Report a bug on GitHub. Detailed instructions on which case to create a problem on GitHub and how to do it can be found in [this article](). 

Reviewer 3: ```suggestion
1.  Report a bug on GitHub. Detailed instructions on creating an issue on GitHub can be found in [this article](). 
```

2.  Send an archive with logs and playback time to `devteam@adguard.com` and attach a link to your issue on GitHub or add an archive with logs and playback time to Google Drive and share it with `devteam@adguard.com`. Attach the Google Drive link to your issue on GitHub.

### Firefox

> 1.  Open AdGuard VPN Browser Extension, if possible, repeat the actions that led to the error. Note the exact time when this error occurred. 

Reviewer 3: same about the name of the product. let's be consistent

2.  Go to `about:addons`
3.  Click *Debug Add-ons*

    ![Debug Add-ons *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/add-ons.png)

4.  Click *Inspect*

    ![Inspect *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/inspect.png)

5.  Go to the *Console* tab

    ![Console *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/ff_console.png)

6.  Click *Save all Messages to file*

    ![Save *border](https://cdn.adguardvpn.com/content/kb/vpn/browser_extension/save-to-file.png)

Now that you have collected the logs, you need to hand them over to our development team. To do this: 

> 1.  Report a bug on GitHub. Detailed instructions on which case to create a problem on GitHub and how to do it can be found in [this article](). 

Reviewer 3: ```suggestion
1.  Report a bug on GitHub. Detailed instructions on creating an issue on GitHub and how to do it can be found in [this article](). 
```

## Collecting and sending logs via the *Report a bug* section

1.  Open AdGuard VPN Browser extension, if possible, repeat the actions that led to the error. Note the exact time when this error occurred.
2.  Open *Settings* by clicking the hamburger menu icon (☰) → *Support* → *Report a bug*.
3.  In the opened form, leave an automatically inserted email address or enter another one and describe the error found, including the time when this error occurred. If you can't reproduce the problem, specify as accurately as possible when it last occurred.
4.  Make sure that there is a check mark next to *Include the diagnostic report in the message*, and tap *Submit*. This way, you will send logs along with the bug report.  

## Collecting and sending logs via the *Export logs* button

1.  Open AdGuard VPN Browser extension, if possible, repeat the actions that led to the error. Note the exact time when this error occurred. 
2.  Right-click the AdGuard VPN Browser extension icon. 
3.  In the opened menu click *Export Logs*.

> 4.  As a result, an archive with logs will be saved to your device in `txt` format.

Reviewer: just a txt file, not in the archive, maybe in future we will fix it and export the archive
